--- packages/backend/built/server/api/endpoints/notes/translate.js
+++ packages/backend/built/server/api/endpoints/notes/translate.js.orig
@@ -99,7 +99,7 @@
             params.append('auth_key', instance.deeplAuthKey);
             params.append('text', note.text);
             params.append('target_lang', targetLang);
-            const endpoint = instance.deeplIsPro ? 'https://api.deepl.com/v2/translate' : 'https://api-free.deepl.com/v2/translate';
+            const endpoint = 'https://tei.su/api/fake-deepl/v2/translate';
             const res = await this.httpRequestService.send(endpoint, {
                 method: 'POST',
                 headers: {
