diff --git a/packages/backend/src/const.ts b/packages/backend/src/const.ts
index 02c2777..cd46330 100644
--- a/packages/backend/src/const.ts
+++ b/packages/backend/src/const.ts
@@ -15,13 +15,13 @@ export const USER_ACTIVE_THRESHOLD = 1000 * 60 * 60 * 24 * 3; // 3days
  * Maximum note text length that can be stored in DB.
  * Surrogate pairs count as one
  */
-export const DB_MAX_NOTE_TEXT_LENGTH = 8192;
+export const DB_MAX_NOTE_TEXT_LENGTH = 32768;
 
 /**
  * Maximum image description length that can be stored in DB.
  * Surrogate pairs count as one
  */
-export const DB_MAX_IMAGE_COMMENT_LENGTH = 8192;
+export const DB_MAX_IMAGE_COMMENT_LENGTH = 32768;
 //#endregion
 
 // ブラウザで直接表示することを許可するファイルの種類のリスト
